internal/storage/location.go

```diff
@@ -12,92 +12,47 @@ import (
 	"github.com/artefactual-sdps/enduro/internal/storage/types"
 )
 
-type InternalLocationFactory func(config *LocationConfig) (Location, error)
-
-var DefaultInternalLocationFactory = func(config *LocationConfig) (Location, error) {
-	return NewInternalLocation(config)
-}
-
-type LocationFactory func(location *goastorage.Location) (Location, error)
-
-var DefaultLocationFactory = func(location *goastorage.Location) (Location, error) {
-	return NewLocation(location)
-}
-
-type Location interface {
+type Locationer interface {
```

Comment:
Effective Go says:
> Interfaces with only one or two methods are common in Go code, and are usually given a name derived from the method, such as io.Writer for something that implements Write.
Ref: https://go.dev/doc/effective_go#interfaces

That doesn't sound like a prohibition using the "-er" suffix for interfaces with more than two methods.  I find it confusing when a concrete noun like "Location" can refer to an struct or an interface.  I think "Locator" would probably be better English, but proper English doesn't seem to be a requirement for interface names either. :laughing:

Reply:
More on that topic here: https://go.dev/doc/effective_go#interface-names.

To be considered: Location is also the name used in Archivematica Storage Service.

Reply:
Nothing in https://go.dev/doc/effective_go#interface-names suggests that you shouldn't use the "-er" naming pattern for interfaces that have more than one method. :confused: 

To clarify, I assume you are saying Location the name of an _interface_ in the Archivematica Storage Service?

Another consideration is that in the Goa `storage` Service, `Location` is the name of a struct type:
https://github.com/artefactual-sdps/enduro/blob/main/internal/api/gen/storage/service.go#L89

Reply:
Also at this point `storage.location` is the only type that implements the `Locationer` interface.  We could just ditch the interface altogether and just use a `storage.Location` concrete type.  The `storage.location` fields aren't exported, so I think the worst thing you could do if it was `storage.Location` (exported) is create an empty struct. :shrug:

Reply:
Another option for the interface name is `LocationProvider`

Reply:
> Also at this point storage.location is the only type that implements the Locationer interface. 

The interface can be useful because you can inject a mock wherever is consumed, and the mock help us build better tests, allows us for testing complex things like retries, rare conditions, error paths, etc... Regardless, the choice is not critical at this point. As you wish! We can revert to the previous approach should it prove necessary in the future.

Reply:
> RE: Locationer | LocationProvider

These two might suggest an entity that performs some action related to locations rather than an entity that represents a location itself, maybe? To be clear I'm indifferent to the final decision and I'd be fine using Locationer, but I must admit that I find enjoyable to think about it. I've been looking for interface names. I've found the `io` package in stdlib interesting: [`File`](https://pkg.go.dev/io/fs#File) or [`FS`](https://pkg.go.dev/io/fs#FS) are some of their choices for interface names. [Upspin](https://pkg.go.dev/upspin.io/upspin) has interfaces like Packer, KeyServer, Client, File, Config, Dialer or Service.

Reply:
Okay, I have come around on the `Location` interface name, and gone back to that and `locationImpl` for the concrete type.  Man, naming right? :face_with_spiral_eyes: :